aws_sagemaker_studio/introduction_to_amazon_algorithms/xgboost_abalone/xgboost_abalone_dist_script_mode.ipynb

```diff
@@ -99,6 +99,8 @@
     "import boto3\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #7.    random.seed(42)how r we using this seed?
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3354/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
We use this seed to set the randomization. So without this seed, the random numbers generated will be different every time. Setting the seed will ensure the same random numbers will be used each time.

https://github.com/dmlc/xgboost/issues/1458 states that the error could be caused by how the data is set up (i.e. randomly split into train and val, which is influenced by the see). After lots of trial and error, I found that it has to do with how our data is split. We use the random package to split our data into train and val set. If I set random.seed(3) , this error can be reproduced. But if I set it another number like random.seed(42) the problem is solved. Originally, a seed wasn't set, which makes sense why the failures are not consistent

Reply:
<p>Got it, thanks!</p>